### PR TITLE
[Dist][Test] Add tests for multi-node DistEmbedding

### PR DESCRIPTION
**Pull request by Kh4L**
## Description
Add tests for `DistEmbedding` and in a multi-node environment.

### CODE REVIEW

File: tests/dist/python/run_dist_objects.py [hunk 1]
```diff
@@ -88,18 +145,21 @@ def test_dist_tensor(g, rank):
...
-    elif target == "DistEmbedding":
-        # TODO: implement DistEmbedding
-        pass
+
+    target_func_map = {"DistTensor": test_dist_tensor,
+                       "DistEmbedding": test_dist_embedding,
+                       }
+
+    target = os.environ.get("DIST_DGL_TEST_OBJECT_TYPE", "")
+    if target not in target_func_map:
```
Comment (Rhett-Ying) [inline tests/dist/python/run_dist_objects.py:157]: why do we need this `if-else`? optionally run some tests during debug? I think it's better to define in `@pytest.mark.parametrize()` or just remove such arg.

**Review by Rhett-Ying — CHANGES_REQUESTED** [review]

File: tests/dist/python/run_dist_objects.py [hunk 2]
```diff
@@ -71,15 +79,64 @@ def dist_tensor_test_persistent(data_shape):
...
+    emb = dgl.distributed.DistEmbedding(num_nodes, 1, name=name, init_func=zeros_init)
+    lr = 0.001
+    optim = dgl.distributed.optim.SparseAdagrad(params=[emb], lr=lr)
+
+    stride = 3
+
+    local_rank = rank % num_client_per_machine
+    idx = F.arange(local_rank*stride, (local_rank+1)*stride)
+
+    if part_id == 0:
```
[the remark this answers is not in the document]
Comment (Kh4L) [reply tests/dist/python/run_dist_objects.py:106]: Similarly to what's done for `DistTensor`, we want to check if the changes are only made by one node and correctly distributed across the nodes `

File: tests/dist/python/run_dist_objects.py (hunk 1 above)
Comment (Kh4L) [reply tests/dist/python/run_dist_objects.py:157]: This file is not run by `pytest`, so we need to do that

File: tests/dist/python/run_dist_objects.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Rhett-Ying) [reply tests/dist/python/run_dist_objects.py:106]: how about `if part_id % 2 == 0`?

**Review by Rhett-Ying — APPROVED** [review]

**Pull request merged**